firewall/privacy_mapper.go

```diff
@@ -812,6 +808,89 @@ func handleUpdatePolicyResponse(db firewalldb.PrivacyMapDB,
 	}
 }
 
+func handleWalletBalanceResponse(_ firewalldb.PrivacyMapDB,
+	flags session.PrivacyFlags,
+	randIntn func(int) (int, error)) func(ctx context.Context,
+	r *lnrpc.WalletBalanceResponse) (proto.Message, error) {
+
+	return func(_ context.Context, r *lnrpc.WalletBalanceResponse) (
+		proto.Message, error) {
+
+		totalBalance, err := maybeHideAmount(
+			flags, randIntn, r.TotalBalance,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		confirmedBalance, err := maybeHideAmount(
+			flags, randIntn, r.ConfirmedBalance,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		unconfirmedBalance, err := maybeHideAmount(
+			flags, randIntn, r.UnconfirmedBalance,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		lockedBalance, err := maybeHideAmount(
+			flags, randIntn, r.LockedBalance,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		reservedBalanceAnchorChan, err := maybeHideAmount(
+			flags, randIntn, r.ReservedBalanceAnchorChan,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		accountBalance := r.AccountBalance
+		if !flags.Contains(session.ClearAccountBalances) {
+			// We remove account information.
+			accountBalance = make(
+				map[string]*lnrpc.WalletAccountBalance,
+				len(r.AccountBalance),
+			)
+		}
+
+		return &lnrpc.WalletBalanceResponse{
+			TotalBalance:       int64(totalBalance),
+			ConfirmedBalance:   int64(confirmedBalance),
+			UnconfirmedBalance: int64(unconfirmedBalance),
+			LockedBalance:      int64(lockedBalance),
+			ReservedBalanceAnchorChan: int64(
+				reservedBalanceAnchorChan,
```

Comment:
i think all these int64 conversions are redundant (the types are already int64s)